pkg/repo/fs.go

```diff
@@ -101,15 +101,11 @@ func (fsr *FsRepo) Init(c config.ReadWriter) error {
 		return err
 	}
 
-	// in the event a user has provided a config without a path we need to set it here base on what the repo was
+	// in the event a user has provided a config without a path we need to set it here based on what the repo was
 	// initialized with.
-	if cfg.Repo == "" {
-		cfg.Repo = fsr.path
-		c.Set("repo", fsr.path)
-	}
-
-	if err := initRepoFiles(cfg); err != nil {
-		return fmt.Errorf("failed to initialize repo: %w", err)
+	if cfg.DataDir == "" {
+		cfg.DataDir = fsr.path
+		c.Set("DataDir", fsr.path)
```

Comment:
Before we can create a config we need to know the repo path. This is because we want to maintain backwards compatibility with existing behavior: When no config is provided check the repo, which may contain a config. See line 71

https://github.com/bacalhau-project/bacalhau/blob/50831fa55c96660fdc573aab4a4441f10d9f6000/cmd/util/repo.go#L58-L101

When we opt to stop searching the repo path for a config file we can remove this logic.

Reply:
I hear you, but I still don't see that a reason for this behaviour and for only setting `cfg.DataDir` with flag provided repo if the `cfg` has no `DataDir`

Take an example if a config was found in the repo, but that config had a `DataDir` configuration set to a different repo path. In the current implementation you will be respecting the `DataDir` in the config file instead of the flag provided `--repo/--data-dir` that the user provided

Lets decouple two things:
1. finding the config file: current implementation looks ok where we also look in the repo dir temporarily. To confirm though, the repo is either provided as a flag, env variable or default value which is currently `~/.bacalhau`, right? 
2. setting repo value in the config file: this should be treated the same as setting other config values that also have cli flags regardless how the config file was found and where it lives. We just need to make sure the logic for finding the repo path is the same when we first tried to look for the config file and when we are setting to set the default value for config